test/chain-adapters/Optimism_Adapter.ts

```diff
@@ -0,0 +1,86 @@
+let l1CrossDomainMessenger: FakeContract, l1StandardBridge: FakeContract;
+
+const optimismChainId = 10;
+
+describe("Optimism Chain Adapter", function () {
+  beforeEach(async function () {
+    [owner, dataWorker, liquidityProvider] = await ethers.getSigners();
+    ({ weth, dai, l2Weth, l2Dai, hubPool, mockSpoke, timer } = await hubPoolFixture());
+    await seedWallet(dataWorker, [dai], weth, amountToLp);
+    await seedWallet(liquidityProvider, [dai], weth, amountToLp.mul(10));
+
+    await enableTokensForLP(owner, hubPool, weth, [weth, dai]);
+    await weth.connect(liquidityProvider).approve(hubPool.address, amountToLp);
+    await hubPool.connect(liquidityProvider).addLiquidity(weth.address, amountToLp);
+    await weth.connect(dataWorker).approve(hubPool.address, bondAmount.mul(10));
+    await dai.connect(liquidityProvider).approve(hubPool.address, amountToLp);
+    await hubPool.connect(liquidityProvider).addLiquidity(dai.address, amountToLp);
+    await dai.connect(dataWorker).approve(hubPool.address, bondAmount.mul(10));
+
+    l1StandardBridge = await createMock("L1StandardBridge");
+    l1CrossDomainMessenger = await createMock("L1CrossDomainMessenger");
+
+    optimismAdapter = await (
+      await getContractFactory("Optimism_Adapter", owner)
+    ).deploy(weth.address, hubPool.address, l1CrossDomainMessenger.address, l1StandardBridge.address);
+
+    await hubPool.setCrossChainContracts(optimismChainId, optimismAdapter.address, mockSpoke.address);
+    await hubPool.whitelistRoute(optimismChainId, weth.address, l2Weth);
+    await hubPool.whitelistRoute(optimismChainId, dai.address, l2Dai);
+  });
+
+  it("Only owner can set l2GasValues", async function () {
+    expect(await optimismAdapter.callStatic.l2GasLimit()).to.equal(sampleL2Gas);
+    await expect(optimismAdapter.connect(liquidityProvider).setL2GasLimit(sampleL2Gas + 1)).to.be.reverted;
+    await optimismAdapter.connect(owner).setL2GasLimit(sampleL2Gas + 1);
+    expect(await optimismAdapter.callStatic.l2GasLimit()).to.equal(sampleL2Gas + 1);
+  });
+  it("Correctly calls appropriate Optimism bridge functions when making ERC20 cross chain calls", async function () {
+    // Create an action that will send an L1->L2 tokens transfer and bundle. For this, create a relayer repayment bundle
+    // and check that at it's finalization the L2 bridge contracts are called as expected.
+    const { leafs, tree, tokensSendToL2 } = await constructSingleChainTree(dai, 1, optimismChainId);
+    await hubPool.connect(dataWorker).initiateRelayerRefund([3117], 1, tree.getHexRoot(), mockTreeRoot);
+    await timer.setCurrentTime(Number(await timer.getCurrentTime()) + refundProposalLiveness);
+    await hubPool.connect(dataWorker).executeRelayerRefund(leafs[0], tree.getHexProof(leafs[0]));
+
+    // The correct functions should have been called on the optimism contracts.
+    expect(l1StandardBridge.depositERC20To).to.have.been.calledOnce; // One token transfer over the bridge.
```

Comment:
this syntax is TOO CLEAN. Do we get this because `smock` extends the chai `expect` module?

Reply:
yup :)